### PR TITLE
chore: remove dropzone pacakge

### PR DESCRIPTION
**Pull request by juliusmarminge**
- remove dropzone package
- move `@uploadthing/dropzone/core` into `@uploadthing/shared`
- move framework specific code into the framework package

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `file-selector` dependency for improved file handling.

- **Bug Fixes**
	- Removed deprecated `@uploadthing/dropzone` dependency to streamline package management and resolve bundling issues.

- **Refactor**
	- Enhanced the dropzone component's functionality and usability in Solid.js with improved state management and event handling.

- **Documentation**
	- Updated package metadata to reflect changes in dependencies and functionalities.

### CODE REVIEW

File: packages/shared/src/dropzone-utils.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 
-import type { AcceptProp, DropzoneState } from "./types";
+export type DropzoneState = {
+  isFocused: boolean;
+  isDragActive: boolean;
+  isDragAccept: boolean;
+  isDragReject: boolean;
+  isFileDialogActive: boolean;
+  acceptedFiles: File[];
+};
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/shared/src/dropzone-utils.ts:20]: good potential feature to add in a follow-up. cc @markflorkowski

File: packages/solid/src/components/dropzone.tsx [hunk 2]
```diff
@@ -305,3 +325,261 @@ export const UploadDropzone = <
...
+  };
+
+  const onFocus = () => setState("isFocused", true);
+  const onBlur = () => setState("isFocused", false);
+  const onClick = () => () => {
+    // In IE11/Edge the file-browser dialog is blocking, therefore, use setTimeout()
+    // to ensure React can handle state changes
+    // See: https://github.com/react-dropzone/react-dropzone/issues/450
+    isIeOrEdge() ? setTimeout(openFileDialog, 0) : openFileDialog();
+  };
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/solid/src/components/dropzone.tsx:544]: part of fxiing the dropzones in #980

File: packages/react/src/components/dropzone.tsx [hunk 3]
```diff
@@ -376,3 +400,316 @@ export function UploadDropzone<
...
+  }, []);
+
+  // Update focus state for the dropzone
+  const onFocus = useCallback(() => dispatch({ type: "focus" }), []);
+  const onBlur = useCallback(() => dispatch({ type: "blur" }), []);
+
+  const onClick = useCallback(() => {
+    // In IE11/Edge the file-browser dialog is blocking, therefore,
+    // use setTimeout() to ensure React can handle state changes
+    isIeOrEdge() ? setTimeout(openFileDialog, 0) : openFileDialog();
```
[the remark this answers is not in the document]
Comment (markflorkowski) [reply packages/react/src/components/dropzone.tsx:655]: Honestly, yeah I'm down. Cc @t3dotgg

**Review by markflorkowski — APPROVED** [review]
Can roll the removing ie11 check into the other fix PR

**Pull request merged**